ApiDocs.DocumentationGeneration/Model/DocumentationComplexType.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * Markdown Scanner
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved. 
+ * 
+ * MIT License
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of 
+ * this software and associated documentation files (the ""Software""), to deal in 
+ * the Software without restriction, including without limitation the rights to use, 
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so, 
+ * subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all 
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, 
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A 
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT 
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION 
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE 
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+namespace ApiDocs.DocumentationGeneration.Model
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using ApiDocs.DocumentationGeneration.Extensions;
+    using ApiDocs.Validation.OData;
+
+    using Newtonsoft.Json;
+
+    public class DocumentationComplexType
+    {
+        public DocumentationComplexType(EntityFramework entityFramework, ComplexType complexType)
```

Comment:
This is a fine place to start, but I think in the long run we should merge Documentation*Type and the stuff we define in the Validation model, so we have one "uber" model for documentation elements that we use throughout Markdown Scanner. This change introduces a new (but more specific) model, which I think we'll ultimately want to have everywhere.